bin/remote-prover/README.md

```diff
@@ -68,6 +70,37 @@ make install-remote-prover
 
 The CLI can be installed from the source code using specific git revisions with `cargo install` or from crates.io with `cargo install miden-remote-prover`.
 
+## Client Library
```

Comment:
It feels like we need to re-organize this file to make two top-level sections - client and server. For example, the "Source installation" section above is not really relevant to the client - and for the client, we should have a different section for installation.